src/main/java/com/rapid7/client/dcerpc/msrrp/objects/RPCSecurityDescriptor.java

```diff
@@ -0,0 +1,122 @@
+   [size_is(cbInSecurityDescriptor), length_is(cbOutSecurityDescriptor)]
+     PBYTE lpSecurityDescriptor;
+   DWORD cbInSecurityDescriptor;
+   DWORD cbOutSecurityDescriptor;
+ } RPC_SECURITY_DESCRIPTOR,
+ *  *PRPC_SECURITY_DESCRIPTOR;
+</pre>
+<p><strong>lpSecurityDescriptor:</strong>  A buffer that
+contains a <a href="https://msdn.microsoft.com/en-us/library/cc230366.aspx">SECURITY_DESCRIPTOR</a>,
+as specified in <a href="https://msdn.microsoft.com/en-us/library/cc230273.aspx">[MS-DTYP]</a>
+section 2.4.6.</p>
+<p><strong>cbInSecurityDescriptor:</strong>  The size in
+bytes of the security descriptor.</p>
+<p><strong>cbOutSecurityDescriptor:</strong>  The size
+in bytes of the security descriptor.</p>
+@see <a href="https://msdn.microsoft.com/en-us/library/cc244924.aspx">https://msdn.microsoft.com/en-us/library/cc244924.aspx</a>
+ */
+public class RPCSecurityDescriptor implements Unmarshallable, Marshallable {
+    private int cbInSecurityDescriptor;
+    private int cbOutSecurityDescriptor;
+    private byte[] rawSecurityDescriptor;
+
+    public void setCbInSecurityDescriptor(final int size) {
+        this.cbInSecurityDescriptor = size;
+    }
+
+    public void setCbOutSecurityDescriptor(final int size) {
+        this.cbOutSecurityDescriptor = size;
+    }
+
+    public void setSecurityDescriptor(final byte[] securityDescriptorBuf) {
+        this.rawSecurityDescriptor = securityDescriptorBuf;
+    }
+
+    @Override
+    public void unmarshalPreamble(final PacketInput in) throws IOException {
+    }
+
+    @Override
+    public void unmarshalEntity(final PacketInput in) throws IOException {
+        in.align(Alignment.FOUR);
+        final int refID = in.readReferentID();
+        if (refID != 0) {
+            cbInSecurityDescriptor = in.readInt();
+            cbOutSecurityDescriptor = in.readInt();
+            rawSecurityDescriptor = new byte[cbOutSecurityDescriptor];
+        }
+    }
+
+    @Override
+    public void unmarshalDeferrals(final PacketInput in) throws IOException {
+        if (rawSecurityDescriptor == null)
+            return;
+        in.align(Alignment.FOUR);
+        in.readInt();
+        in.readInt(); // offset
+        in.readInt();
+        // TODO: Return parsed object.
+        in.readRawBytes(rawSecurityDescriptor);
+    }
+
+    @Override
+    public void marshalPreamble(final PacketOutput out) throws IOException {
+    }
+
+    @Override
+    public void marshalEntity(final PacketOutput out) throws IOException {
+        if (rawSecurityDescriptor == null) {
+            out.writeNull();
+        } else {
+            out.writeReferentID();
+        }
+
+        out.writeInt(cbInSecurityDescriptor);
+        out.writeInt(cbOutSecurityDescriptor);
+    }
+
+    @Override
+    public void marshalDeferrals(final PacketOutput out) throws IOException {
+        if (rawSecurityDescriptor == null)
+            return;
+
+        out.writeInt(cbInSecurityDescriptor);
```

Comment:
This would be `cbOutSecurityDescriptor` as well. For conformant arrays, we generally write `MaximumSize` the same as `Size`.

Reply:
In this case, the cbInSecurityDescriptor is what the client asks the server to allocate, the cbOutSecurityDescriptor is the actual size that the server responded with.